src/write-archive/index.test.ts

```diff
@@ -41,6 +42,76 @@ describe('writeTestResult', () => {
     );
   });
 
+  it('successfully generates test results with mapped source entries', async () => {
+    // @ts-expect-error Jest mock
+    fs.ensureDir.mockReturnValue(true);
+    const oldMockFs = fs.outputFile;
+    const storyJson = {
+      childNodes: [
+        {
+          attributes: {
+            src: '/bogano',
+          },
+        },
+      ],
+    };
+
+    const expectedMappedJson = {
+      childNodes: [
+        {
+          attributes: {
+            src: '/coruscant',
+          },
+        },
+      ],
+    };
+
+    const sourceMapping = new Map<string, string>();
+    sourceMapping.set('/bogano', '/coruscant');
+    await writeTestResult(
+      // the default output directory in playwright
+      { title: 'Toy Story', outputDir: resolve('test-results/toy-story-chromium') } as TestInfo,
+      { home: Buffer.from(JSON.stringify(storyJson)) },
+      {
+        'http://localhost:3000/home': {
+          statusCode: 200,
+          body: Buffer.from(JSON.stringify(storyJson)),
+        },
+      },
+      { viewport: { height: 480, width: 720 } },
+      sourceMapping
+    );
+
+    const expectedBuffer = Buffer.from(JSON.stringify(expectedMappedJson));
+    const oldMock = fs.outputFile;
+
+    // XXX_jwir3: Ugh, this is annoying that I have to do it this way, but I think jest is caching the mocks
+    //            somehow. I always get the wrong result when I use expect(fs.outputFile).toHaveBeenCalledWith().
+    fs.outputFile = (filename: string, data: string | Buffer, callback: fs.NoParamCallback) => {
+      expect(data).toEqual(expectedBuffer);
+    };
+
+    expect(fs.ensureDir).toHaveBeenCalledTimes(1);
+    expect(fs.outputJson).toHaveBeenCalledTimes(1);
+    expect(fs.outputJson).toHaveBeenCalledWith(
+      resolve('./test-results/chromatic-archives/toy-story.stories.json'),
+      {
+        stories: [
+          {
+            name: 'home',
+            parameters: {
+              chromatic: { viewports: [720] },
+              server: { id: 'toy-story-home.snapshot.json' },
+            },
+          },
+        ],
+        title: 'Toy Story',
+      }
+    );
+
+    fs.outputFile = oldMock;
```

Comment:
This is being [reset](https://github.com/chromaui/test-archiver/pull/24/files#diff-b6a957839ae7b4419b805f899483d22c465179a81da04f9ff9edb69fce7f7009R10) in the `beforeEach`, so you can remove this one in favor of those (or change which [reset method](https://jestjs.io/docs/mock-function-api#mockfnmockclear) is used).

Reply:
This code has become obsolete because I was able to get the testing to work (see comment below, as a reply to Tom).